### ISSUE TITLE
duplicate names error

### ISSUE THREAD

**Issue by tyronerees** [fitbenchmarking/fitbenchmarking#408]
**Description of the error**
If there are two problems with the same name in the folder being benchmarked, then fitbenchmarking fails with:
```
Traceback (most recent call last):
  File "/home/dgk98065/.local/bin/fitbenchmarking", line 11, in <module>
    load_entry_point('FitBenchmarking', 'console_scripts', 'fitbenchmarking')()
  File "/home/dgk98065/fitbenchmarking/fitbenchmarking/cli/main.py", line 133, in main
    run(problem_sets=args.problem_sets, options_file=args.options_file)
  File "/home/dgk98065/fitbenchmarking/fitbenchmarking/cli/main.py", line 96, in run
    options=options)
  File "/home/dgk98065/fitbenchmarking/fitbenchmarking/core/results_output.py", line 43, in save_results
    create_plots(options, results, best_results, group_name, fig_dir)
  File "/home/dgk98065/fitbenchmarking/fitbenchmarking/core/results_output.py", line 133, in create_plots
    initial_guess_path = plot.plot_initial_guess()
  File "/home/dgk98065/fitbenchmarking/fitbenchmarking/results_processing/plots.py", line 128, in plot_initial_guess
    ini_guess = self.problem.starting_values[self.count - 1].values()
IndexError: list index out of range

```

**Describe the expected result**
Ideally the output should still be produced, possibly by appending something to the name.  If this is not possible, then a human readable error message should be raised (ideally right at the start, so that data isn't lost)

Problems that trigger this are here: [duplicate_name_bug_example.zip](https://github.com/fitbenchmarking/fitbenchmarking/files/4107493/duplicate_name_bug_example.zip)
.

**Comment by AndrewLister-STFC** [comment]
This will be fixed by #4, as I had to change how names are used to get multiple data files into the tables.
It's not entirely foolproof as I thing having 3 files (`test`, `test`, `test 1`) with identical names apart from 1 has an index will cause it to break, but I think thats unlikely.

**Comment by AndrewLister-STFC** [comment]
This was fixed at one point...

It looks like it got removed in [`55e009e` (#527)](https://github.com/fitbenchmarking/fitbenchmarking/pull/527/commits/55e009e6635191c5b0f3b0eb216f720237dea9d4) due to leaving an extra `1` next to unique names... we should review this.
